frontend/ui/uimanager.lua

```diff
@@ -660,6 +663,31 @@ function UIManager:resetInputTimeout()
     self.INPUT_TIMEOUT = nil
 end
 
+function UIManager:handleInputEvent(input_event)
+    if input_event.handler ~= "onInputError" then
+        self.event_hook:execute("InputEvent", input_event)
+    end
+    local handler = self.event_handlers[input_event]
+    if handler then
+        handler(input_event)
+    else
+        self.event_handlers["__default__"](input_event)
+    end
+end
+
+-- Process all pending events on all registered ZMQs.
+function UIManager:processZMQs()
+    for _, zeromq in ipairs(self._zeromqs) do
+        for input_event in zeromq.waitEvent,zeromq do
+            self:handleInputEvent(input_event)
+        end
+        -- input_event = zeromq:waitEvent()
```

Comment:
What's this commented out stuff for exactly? It looks like it should probably just be removed?

Reply:
Done.